static/js/builder.js

```diff
@@ -1,56 +1,34 @@
 "use strict";
 
-// These could/should be passed to CodelistBuilder as props
-const SEARCHES = JSON.parse(document.getElementById("searches").textContent);
-const FILTER = JSON.parse(document.getElementById("filter").textContent);
-const CODE_TO_STATUS = JSON.parse(
-  document.getElementById("code-to-status").textContent
-);
-const TABLES = JSON.parse(document.getElementById("tables").textContent);
-const ANCESTORS_MAP = JSON.parse(
-  document.getElementById("ancestors-map").textContent
-);
-const DESCENDANTS_MAP = JSON.parse(
-  document.getElementById("descendants-map").textContent
-);
-const CODES = Object.keys(ANCESTORS_MAP);
-const IS_EDITABLE = JSON.parse(
-  document.getElementById("isEditable").textContent
-);
-const UPDATE_URL = JSON.parse(
-  document.getElementById("update-url").textContent
-);
-const SEARCH_URL = JSON.parse(
-  document.getElementById("search-url").textContent
-);
-
 class CodelistBuilder extends React.Component {
   constructor(props) {
     super(props);
 
-    this.state = { updateQueue: [], updating: false, isEditable: IS_EDITABLE };
+    this.state = { updateQueue: [], updating: false };
 
-    TABLES.forEach((table) =>
-      table.rows.forEach((row) => {
-        this.state["status-" + row.code] = row.status;
-      })
-    );
-
-    CODES.forEach((code) => {
-      this.state["status-" + code] = CODE_TO_STATUS[code];
+    this.codes = Object.keys(props.ancestorsMap);
+    this.codes.forEach((code) => {
+      this.state["status-" + code] = props.codeToStatus[code];
       this.state["expanded-" + code] = true;
     });
 
     this.updateStatus = this.updateStatus.bind(this);
     this.toggleVisibility = this.toggleVisibility.bind(this);
+    this.getStatus = this.getStatus.bind(this);
+    this.getHasDescendants = this.getHasDescendants.bind(this);
+    this.getIsExpanded = this.getIsExpanded.bind(this);
+    this.getIsVisible = this.getIsVisible.bind(this);
```

Comment:
You can avoid this step by doing `someProp={() => this.someMethod()}` at the call site.

Reply:
Doesn't this create a new function on every render?

Reply:
[Apparently so](https://reactjs.org/docs/faq-functions.html#arrow-function-in-render), TIL!